### DISCUSSION TITLE
How do I change the label background color / text entry box font color?

### DISCUSSION THREAD

**Discussion by tourmalinec** [rdbende/Azure-ttk-theme#49] [Question]
I like your themes.  However,  after setting the current theme to an Azure theme,  if I switch to a different none-Azure ttk theme, the foreground/background colors of labels (TLabel) and foreground color of text input box (TEntry) remain to be the last Azure theme color style.  Is there anyway to reset/change these settings?

**Comment by rdbende** [comment]
I could add a helper function to reset the colors, however the problem is, which colors to reset to? Should it be the previous theme? Should it be the next theme? (How to know, what will be the next theme? And why it doesn't configure the colors for itself?)
Ideas are welcome!

Maybe a better solution would be to explicitly use the `tk_setPalette()` method, and `ttk.Style().configure(".", ...)` to (re)set the appropriate colors.

**Reply by tourmalinec** [reply]
Thanks for replying.  Let me explain my issue:
I have tried to switch themes from Azura themes to ttk themes. 
e.g. When set the asure theme, I call:
`self.tk.call("set_theme", "dark")`
Then switch to a ttk theme:
`self.set_theme("vista")`

This is how my gui looks like in Azure dark theme:
![azure_dark](https://user-images.githubusercontent.com/47292856/192066204-09bba776-9399-4e93-8e1a-2dd27f59a0e0.jpg)
This is how the 'vista' theme looks like (If not switched from azure theme):
![vista](https://user-images.githubusercontent.com/47292856/192066291-c2619c34-18fc-4095-999c-16296111e520.jpg)

But after I switch to 'vista' from Azure dark theme, this is how it looks like:

![vista_mixed](https://user-images.githubusercontent.com/47292856/192066353-2a4d84fa-0efa-4c48-8ba8-850efa1f892a.jpg)

As you can see, the label (TLabel) or entry (TEntry) foreground colors remain as white; the background color of TLabel is still black; the background color of TEntry has been changed to white (vista default), however, due to the foreground color is still white, you won't see the texts unless highlight it.

This does not just happen to 'vista' theme.  For all ttk themes, I can freely switch them without issues and no explicitly color configurations needed.  However, once I used azure themes, this issue persists no matter which ttk theme I switch to (except azure themes itself).  I have tried to use ttk.Style().configure to set colors after switched from azure themes to a ttk theme:
    "background": "#ffffff",
    "foreground": "#000000",
    "fieldbackground": "#ffffff"
but this has no effect.  So I wonder how I can change the colors after switched from azure themes?

**Comment by rdbende** [comment]
I discussed this with Tcl/Tk developers, and the solution will probably be to remove the `tk_setPalette` call, as it's virtually impossible to reset the colors. This has the side effect that the plain Tk widgets won't follow the color scheme, but that's probably acceptable, since they can be themed separately if one really needs to use such widgets.
